integration-tests/test_status.py

```diff
@@ -0,0 +1,42 @@
+import contextlib
+import pytest
+from time import sleep
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_status_registered(external_candlepin, insights_client):
+    """
+    Test `insights-client --status` when the client is registered;
+    in case of legacy_upload status command returns a different output.
+
+    """
+    insights_client.register()
+    assert insights_client.is_registered
+    # Adding a small wait to ensure inventory is up-to-date
+    sleep(5)
+    registration_status = insights_client.run("--status")
+    if insights_client.config.legacy_upload:
+        assert registration_status.returncode == 0
+        assert b"Insights API confirms registration." in registration_status.stdout
+    else:
+        assert b"This host is registered.\n" == registration_status.stdout
+
+
+def test_status_unregistered(external_candlepin, insights_client):
+    """
+    Test `insights-client --status` when the client is unregistered;
+    in case of legacy_upload status command returns a different output.
+
+    """
+    # running registration to ensure system is unregistered
+    with contextlib.suppress(Exception):
+        insights_client.unregister()
+    registration_status = insights_client.run("--status")
```

Comment:
I notice that the return code of `insights-client --status` in legacy upload mode is 1, and that means the `run()` will fail; since the exit code is expected, then I'd say to pass `check=False` as parameter of `run()` here ...